mxcube3/routes/signals.py

```diff
@@ -129,7 +138,7 @@ def loaded_sample_changed(sample):
     else:
         address = ''
         barcode = ''
-
+    mxcube.shapes.clear_all()
```

Comment:
I've guess you've added this because you are mounting samples from outside of MXCuBE ?, the way this is handled today is in mount_sample_cleanup. It turns out that the different sample changers fire this event at different moments which sometimes causes some unexpected behavior.

Its fine to keep it like this, if you are mounting/un-mounting samples from other software often. Otherwise I would remove it and rely on the behavior in mount_sample_cleanup.

Another alternative is to move some of mount_sample_cleanup to this method, but this might become more complicated than we want because of what I mentioned above.

Reply:
we only use mxcube ... for everything!

But I still have seen situations where not all the samples were deleted after a sample mount, thats why I ensure the deletion of old shapes here. Have you seen anything like this?

Reply:
Hm, I have to say that we are not seeing that. The most elegant solution would be if this was handled in the loaded_sample_changed but its for the moment handled in mount_sample_cleanup which is completely fine as long as one assumes that we are only loading samples within MXCuBE. 

So all the clean up associated with mounting a sample is is handled in mount_sample_cleanup, hence its name :). I guess something else is going wrong before clear_all is called if the shapes are not removed ? Its maybe worth looking into ...

Reply:
I could not find the issue other than this line :(

Reply:
Well something is clearly going wrong if the shapes are not removed at the end of mount_sample_cleanup, how are you mounting the sample ?

Reply:
ahh, perfect !